src/Microsoft.Azure.WebJobs/ConnectionProviderAttribute.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Azure.WebJobs
```

Comment:
>WebJobs [](start = 26, length = 7)

Nit: I think we should move these (ConnectionProviderAttr, IConnectionProvider) out of Microsoft.Azure.WebJobs namespace and one level deep, probably Microsoft.Azure.WebJobs.Config. 
User code won't refer to these; so it pollutes their namespace. 
We could apply that rule to any "meta" attribute used to define the binding attributes (AutoResolve, AppSetting, etc). - I should do something similar in my byob PR.